From: Hans de Goede <johannes.goede@oss.qualcomm.com>
Date: Sun, 16 Aug 2026 14:24:57 +0200
Subject: fix(devicetree-firmware): include Qualcomm X2 laptop model specific
 firmwares

Qualcomm laptops typically have laptop model specific firmwares.

Extend install_generic() to pick up the laptop model specific firmwares
for the new X2 / Glymur laptops which have a filename pattern of:
<fwdir>/qcom/glymur/<vendor>/<model>/*.[mbn|elf]

Origin: upstream, https://github.com/dracut-ng/dracut/commit/911aeed0a2bc2c59fcd62f47cf6a1cb7354f08b1
---
 modules.d/70devicetree-firmware/module-setup.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/70devicetree-firmware/module-setup.sh b/modules.d/70devicetree-firmware/module-setup.sh
index e725ac9..ceb4958 100755
--- a/modules.d/70devicetree-firmware/module-setup.sh
+++ b/modules.d/70devicetree-firmware/module-setup.sh
@@ -49,7 +49,7 @@ install_generic() {
     # shellcheck disable=SC2154 # fw_dir is set by dracut.sh
     for _fwdir in $fw_dir; do
         # add laptop model specific firmwares for qcom WoA laptops
-        for _soc in qcom/sc8280xp qcom/x1e80100; do
+        for _soc in qcom/sc8280xp qcom/x1e80100 qcom/glymur; do
             # add '*' after mbn, elf for .gz, etc. compression
             for _fw in "$_fwdir/$_soc"/*/*/*.mbn* "$_fwdir/$_soc"/*/*/*.elf*; do
                 [ -f "$_fw" ] || continue
